src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -88,6 +100,46 @@ module.exports = function<T, P, I, TI, C, CX>(
     updateClassInstance,
   } = ReactFiberClassComponent(scheduleUpdate, getPriorityContext);
 
+  function isValidNode(node) {
```

Comment:
I think a good guideline is that we shouldn't have *any* production invariants unless the branch is already covered regardless.